### ISSUE TITLE
Get more info from splaxi about our help

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#4258]
https://twitter.com/splaxi/status/1053986222589313024

![image](https://user-images.githubusercontent.com/8278033/47266968-cf279800-d53d-11e8-8365-6e165fcc7642.png)

His pester tests show things that need to be fixed, awesome! 🔥 🔥🔥🔥 🚒

**Comment by Splaxi** [comment]
Specific function that I don't understand how to solve.

```
Example - Remove-DbaDbMasterKey -SqlInstance Server1"
Parameter set cannot be resolved using the specified named parameters.
```

I can't see if the default parameterset name is correct or not.

**Comment by Splaxi** [comment]
Specific function that I don't understand how to solve.

```
Read-DbaBackupHeader -Path https://dbatoolsaz.blob.core.windows.net/azbackups/restoretime/restoretime_201705131850.bak -AzureCredential AzureBackupUser
Cannot process argument transformation on parameter 'SqlInstance'. Cannot convert value "" to type "Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter". Error: "Please provide an instance name
```

I can't validate if it is expected to actually work **without** an instance or we need to update the example.

**Comment by potatoqualitee** [comment]
that last one, there is no SqlInstance, but it's not mandatory (which isn't unusual, as we dont check because of piped input)

**Comment by potatoqualitee** [comment]
Remove-DbaDbMasterKey is all messed up with param sets, I'll figure this out in the code thank you. We prefer informing users of their mistake over using param sets cuz of issues like this. I have no idea what's wrong here - will fix both command and example 👍

**Comment by joshcorr** [comment]
`$script:Instance1` is found in the  `.\tests\constants.ps1` file. It is used for running the Pester tests on appveyor. Usually when running these tests in your own environment you need to change this to reflect your test instances.

**Comment by Splaxi** [comment]
That value is found in an example inside the `Backup-DbaDatabase` function.

I can't say whether or not that is intended - I'm just saying that the example is failing my testing ;)

**Comment by Splaxi** [comment]
Same as earlier.

`-SqlInstance $script:instance2`
```
$results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific -queryname 'Database-scoped Configurations' -databasename $database

Cannot bind argument to parameter 'SqlInstance' because it is null.
```

The example comes from inside `Invoke-DbaDiagnosticQuery`

**Comment by potatoqualitee** [comment]
thanks to you both. Those can be updated to  `sql2017` or whatever you wish, both are valid, tho I see what youre saying.

**Comment by Splaxi** [comment]
So most errors are gone. Like I stated earlier, the gist link contains the current code. I have implemented it in my build / validate pipeline. Do the same if you want. There is an exclude array that will help you skip those special cases that will fail when running the test.

The perfect result should be green and yellow only. Anything red should either be fixed or excluded using the array in question.
